src/sentry/grouping/enhancer/__init__.py

```diff
@@ -145,10 +206,60 @@ def apply_modifications_to_frame(
         extra_fingerprint: str = "",
     ) -> None:
         """This applies the frame modifications to the frames itself. This does not affect grouping."""
-        in_memory_cache: dict[str, str] = {}
 
         # Matching frames are used for matching rules
         match_frames = [create_match_frame(frame, platform) for frame in frames]
+
+        rust_enhanced_frames = None
+        try:
+            use_rust_enhancements = random.random() < options.get(
+                "grouping.rust_enhancers.modify_frames_rate"
+            )
+        except Exception:
+            use_rust_enhancements = False
+        if (enhancements := self.rust_enhancements) and use_rust_enhancements:
+            e = exception_data or {}
+            e = {
+                "ty": e.get("type"),
+                "value": e.get("value"),
+                "mechanism": get_path(e, "mechanism", "type"),
+            }
+            for key in e.keys():
+                value = e[key]
+                if isinstance(value, str):
+                    e[key] = value.encode("utf-8")
+
+            try:
+                rust_enhanced_frames = enhancements.apply_modifications_to_frames(
+                    iter(match_frames), e
+                )
+                metrics.incr("rust_enhancements.modifications_run")
+            except Exception:
+                logger.exception("failed running Rust Enhancements modifications")
+
+            # TODO: once ready, update the frame directly
+            # for frame, (category, in_app) in zip(frames, rust_enhanced_frames):
+            #     if in_app is not None:
+            #         set_in_app(frame, in_app)
+            #     if category is not None:
+            #         set_path(frame, "data", "category", value=category)
+
+        def compare_rust_enhancers():
```

Comment:
_If_ possible, please avoid this function within `compare_rust_enhancers` as it's quite busy as-is.